runners/flink/flink_runner.gradle

```diff
@@ -314,12 +316,20 @@ def createValidatesRunnerTask(Map m) {
         excludeTestsMatching 'org.apache.beam.sdk.testing.TestStreamTest.testFirstElementLate'
         // https://github.com/apache/beam/issues/20844
         excludeTestsMatching 'org.apache.beam.sdk.testing.TestStreamTest.testLateDataAccumulating'
+        if (!config.streaming) {
+          // FlinkBatchExecutionInternalTimeService does not support timer registration on timer firing.
+          excludeTestsMatching 'org.apache.beam.sdk.transforms.ParDoTest$TimestampTests.testOnTimerTimestampSkew'
+        } else {
+          // https://github.com/apache/beam/issues/25485
+          excludeTestsMatching 'org.apache.beam.sdk.transforms.GroupByKeyTest$BasicTests.testAfterProcessingTimeContinuationTriggerUsingState'
```

Comment:
is this still necessary (test still flaky)?  Does this PR make the test more flaky?

Reply:
I tried all the tests in `org.apache.beam.sdk.transforms.GroupByKeyTest$BasicTests`  and `org.apache.beam.sdk.transforms.ParDoTest$TimestampTests` 10 times for each validation (`validatesRunnerBatch`, `validatesRunnerBatchWithDataStream` and `validatesRunnerStreaming`) .


`testOnTimerTimestampSkew` never passes with `validatesRunnerBatchWithDataStream` which I think explains why this test is disabled. `GroupByKeyTest$BasicTests` on the other end always passes so I think this one does not need to be excluded.

Reply:
I think the reason `testOnTimerTimestampSkew` fails when using the datastream API for batch is [FLINK-18647](https://issues.apache.org/jira/browse/FLINK-18647). I don't know how to fix it though. Let me know if you have ideas.